azurelinuxagent/common/protocol/util.py

```diff
@@ -227,41 +223,17 @@ def _get_protocol(self):
         """
         Get protocol instance based on previous detecting result.
         """
-        protocol_file_path = self._get_protocol_file_path()
-        if not os.path.isfile(protocol_file_path):
-            raise ProtocolNotFoundError("No protocol found")
 
         endpoint = self.get_wireserver_endpoint()
         return WireProtocol(endpoint)
 
-    def _save_protocol(self, protocol_name):
-        """
-        Save protocol endpoint
-        """
-        protocol_file_path = self._get_protocol_file_path()
-        try:
-            fileutil.write_file(protocol_file_path, protocol_name)
-        except (IOError, OSError) as e:
-            logger.error("Failed to save protocol endpoint: {0}", e)
-
     def clear_protocol(self):
         """
         Cleanup previous saved protocol endpoint.
         """
         logger.info("Clean protocol and wireserver endpoint")
         self._clear_wireserver_endpoint()
```

Comment:
This will fall back to KNOWN_WIRESERVER_IP and therefore never fail.  There are two choices here

1. Remove this code and related code.
2. Keep this code but always use the "well known" wireserver IP address which totally makes all the DHCP work we did obsolete...